<?xml version="1.0"?>
<!--XML-->
<launch>
    <!-- SETUP ARGUMENTS -->
    <!-- enable/disable running rviz in launcher -->
    <arg name="rviz" default="true" />
    <arg name="camera_ns" default="i3dr_titania" />

    <!-- Reduce point cloud size with VoxelGrid -->
    <node name="reduce_points" pkg="i3dr_detect_3d" type="reduce_points" output="screen" >
        <param name="grid_size" type="double" value="0.005" />
        <remap from="/points2" to="$(arg camera_ns)/points2" /> 
        <remap from="/points2_reduced" to="$(arg camera_ns)/points2_reduced" />
    </node>

    <!-- Detect primative in point cloud -->
    <node name="detect_primitives" pkg="i3dr_detect_3d" type="detect_primitives" output="screen" >
        <remap from="/points2" to="$(arg camera_ns)/points2_reduced" />
    </node>

    <!-- DISPLAY -->
    <!-- run rviz with config file -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rivz" 
        args="-d $(find i3dr_detect_3d)/rviz/titania_prim.rviz"/>

</launch>